File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        l = r = 0

        while r < n:
            nums[l] = nums[r]
            while r < n and nums[l] == nums[r]:
                r += 1
            l += 1
        return l

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> numCount = new HashMap<>();

        PriorityQueue<Map.Entry<Integer, Integer>> maxHeap = new PriorityQueue<>(
            (a, b) -> b.getValue() - a.getValue()
        );

        for (Integer n: nums) {
            numCount.merge(n, 1, Integer::sum);
        }

        maxHeap.addAll(numCount.entrySet());

        int[] result = new int[k];

        for (int i = 0; i < k; i++) {
            result[i] = maxHeap.poll().getKey();
        }

        return result;
    }
}
